src/earn/EarnPoolInfoScreen.tsx

```diff
@@ -683,6 +765,140 @@ function InfoBottomSheet({
   )
 }
 
+function BeforeDepositBottomSheet({
+  forwardedRef,
+  token,
+  canSameChainSwapToDeposit,
+  canCrossChainSwapToDeposit,
+  canAdd,
+  exchanges,
+}: {
+  forwardedRef: RefObject<BottomSheetModalRefType>
+  token: TokenBalance
+  canSameChainSwapToDeposit: boolean
+  canCrossChainSwapToDeposit: boolean
+  canAdd: boolean
+  exchanges: ExternalExchangeProvider[]
+}) {
+  const { t } = useTranslation()
+
+  const actions = useMemo(() => {
+    const visibleActions: AddAssetsAction[] = []
```

Comment:
I think that this is a rather difficult way to implement these actions and a very difficult way for them to be read. Its difficult to see without a lot of effort which actions should be showing up for a given ui. 

It may be eaiser to just hardcode which actions should be visible in the different views that product has decided upon.
```
const visibleActions: AddAssetsAction[] = []

switch(view) {
  case DepositActionView.TokensOnSameChain: 
    visibleActions=[
      getSwapAndDepositAction(),
      getCrossChainSwapAction({tokens, description}),
      getAddAction({tokens, networkId})
    ]
  case DepositActionView.TokensOnDifferentChain:
    // next set of actions
  case DepositActionView.NoTokens: 
    // next set of actions
}
```

Reply:
also note my suggestion to put the description here like i mentioned in my other comment

Reply:
do you mean that I could include the description in the visibleActions list? then it seems like I would need some extra logic for it still in the AddAssetsBottomSheet because the actions are rendered as touchables with icons and we wouldn't want that for the description

Reply:
Or are you suggesting add a description field to the action object that could be rendered before the action touchable in the bottom sheet?

Reply:
Im suggesting that you make an action item a view that includes an (optional) description and a touchable

Reply:
Instead of reusing the `AddAssetsBottomSheet` component, I wonder if this should be it's own bottom sheet. Soma actions here aren't really add actions and the description in the middle is custom enough that it can be it's own component. Maybe we can make a reusable `ActionCard` component that can be shared with `AddAssetsBottomSheet`, `TokenDetailsMoreActions` and here though, but maybe not worth it.